File: TheBossInJS/Gangster.js
var assert = require('assert');
var GANG_TYPE = require('./gangster_type');

function Gangster(type) {

    this.type = type;
    this.pauseHands = 0;
}

Gangster.prototype.imprison = function() {

    this.pauseHands = 2;
}

Gangster.prototype.hospitalize = function() {

    this.pauseHands = 1;
}

Gangster.prototype.inSanction = function() {

    return (this.pausehands > 0);
}

Gangster.prototype.impose = function() {

    assert(this.pauseHands > 0);
    this.pauseHands--;
}

function GangFactory() {}
GangFactory.prototype.createGangster = function(type) {

    if (type !== GANG_TYPE.EXPERT &&
        type !== GANG_TYPE.OCCASIONAL)
        throw "NOT a valid gangster type:" + type;

    return new Gangster(type);
}

module.exports = new GangFactory();

File: TheBoss/src/model/impl/Gangster.java
package model.impl;

import model.constant.GANGSTER_TYPE;

public class Gangster {
	
	private GANGSTER_TYPE type;
	private int pauseHands;
	
	public Gangster(GANGSTER_TYPE type){
		
		this.type = type;
		pauseHands = 0;
	}
	
	public GANGSTER_TYPE getType(){ return type; }	
	public void imprison(){ pauseHands = 2;	}
	public void hostpitalize(){ pauseHands = 1;	}
	public boolean inSanction(){ return (pauseHands > 0); }
	
	public void impose(){
		
		assert(pauseHands > 0);
		pauseHands--;
	}	
}
